### PR TITLE
Added example Account derive from mnemonic in Go

### PR DESCRIPTION
**Pull request by mstrug-rdx**
Implementation based on https://github.com/radixdlt/wallet-compatible-derivation.

### CODE REVIEW

File: radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Example of account derive from mnemonic in Go
+
+Compatible environment: MacOS aarch64.
```
Comment (iamyulong) [inline radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/README.md:3]: Maybe rephrased it as "Tested on MacOS xx.11 with aarch64" since it supports more than MacOS.

File: radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/main.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+func assert(condition bool, message string) {
+    if !condition {
+        panic(message)
+    }
+}
+
+// networkId: 1 - Mainnet, 2 - Stokenet, ...
+// mnemonic: words list space separated
+// passphrase: optional
```
Comment (iamyulong) [inline radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/main.go:22]: What does optional mean? Empty string? Sorry, no prior Rust experience.

**Review by iamyulong — APPROVED** [review]
LGTM

File: radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/main.go (hunk 2 above)
Comment (mstrug-rdx) [reply radix-engine-toolkit/examples/transactions/account-derive-from-mnemonic/Go/main.go]: I've updated that comment to: `passphrase: for Wallet compatible account use empty string`
